src/IECore/StreamIndexedIO.cpp

```diff
@@ -764,6 +728,82 @@ class StreamIndexedIO::Node
 		DirectoryNode *m_node;
 };
 
+//! Small scoped class to read from a given data block in a file
+//! decompressing if required
+class StreamIndexedIO::Reader
```

Comment:
Can this whole commit be squashed into deb4d7677da3618de5d972be333f0f0eb6e9d931? It would make for an easier review of the final PR.